library/include/rocwmma/internal/b32_util.hpp

```diff
@@ -0,0 +1,161 @@
+
+            static_assert(decltype(rIt)::range() == decltype(wIt)::range(),
+                          "Unexpected iterator range mismatch");
+
+            static_assert(SelectIdx < (sizeof(ExtractedT) / sizeof(EmplacedT)),
+                          "Invalid index selection");
+
+            for(uint32_t i = 0u; i < decltype(rIt)::range(); i++, rIt++, wIt++)
+            {
+                FootSpace a;
+                a.extracted  = get<0>(*rIt);
+                get<0>(*wIt) = a.emplaced[SelectIdx];
+            }
+            return result;
+        }
+    };
+
+    template <>
+    struct b32Util<float64_t>
+    {
+        using ExtractedT = float32_t;
+        using EmplacedT  = float64_t;
+
+        using FootSpace = union
+        {
+            ExtractedT extracted[2];
+            EmplacedT  emplaced;
+        };
+
+        template <uint32_t VecSize, uint32_t SelectIdx = 0u>
+        __device__ static auto extract(VecT<EmplacedT, VecSize> const& v)
+        {
+            const uint32_t inflatedSize = VecSize << 1;
+
+            VecT<ExtractedT, inflatedSize> result;
+            auto const                     rIt = makeVectorIterator(v).begin();
+            auto                           wIt = makeVectorIterator(result).begin();
+
+            static_assert(decltype(rIt)::range() == (decltype(wIt)::range() >> 1),
+                          "Unexpected iterator range mismatch");
+
+            static_assert(SelectIdx < (sizeof(EmplacedT) / sizeof(ExtractedT)),
+                          "Invalid index selection");
+
+            for(uint32_t i = 0u; i < decltype(rIt)::range(); i++, rIt++, wIt++)
+            {
+                FootSpace a;
+                a.emplaced         = get<0>(*rIt);
+                get<0>(*wIt)       = a.extracted[0];
+                get<VecSize>(*wIt) = a.extracted[1];
+            }
+            return result;
+        }
+
+        template <uint32_t VecSize, uint32_t SelectIdx = 0u>
+        __device__ static auto emplace(VecT<ExtractedT, VecSize> const& v)
+        {
+            const uint32_t inflatedSize = VecSize << 1;
+
+            VecT<EmplacedT, VecSize> result;
+            auto const               rIt = makeVectorIterator(v).begin();
+            auto                     wIt = makeVectorIterator(result).begin();
+
+            static_assert((decltype(rIt)::range() >> 1) == decltype(wIt)::range(),
+                          "Unexpected iterator range mismatch");
+
+            static_assert(SelectIdx < (sizeof(EmplacedT) / sizeof(ExtractedT)),
+                          "Invalid index selection");
+
+            for(uint32_t i = 0u; i < decltype(wIt)::range(); i++, rIt++, wIt++)
+            {
+                FootSpace a;
+                a.extracted[0] = get<0>(*rIt);
+                a.extracted[1] = get<VecSize>(*rIt);
+                get<0>(*wIt)   = a.emplaced;
+            }
+            return result;
+        }
+    };
```

Comment:
At this point, it's a good question to ask: once we extract both uppers and lowers into their own groups, what is a good way to recombine the registers in 'emplace' such that their f64 order is preserved?

The solution might be some clever register re-ordering among the two groups, or we can look at adding an additional signature for 'emplace':
```
ROCWMMA_DEVICE static auto emplace(VecT<EmplacedT, VecSize> &dst, VecT<ExtractedT, VecSize> const& v )
{
...
    for(i=0...range()){
        FootSpace a = dst[i];
        a.extracted[SelectIdx] = v[i];
        }
}
```
The above will allow you to modify an existing vector dst, which facilitates data recombination.
At the end of the day, what a b32 op might look like for f64:
- Extract upper, lower f32 vecs
- Mirror b32 op on upper and lower f32 vecs
- Emplace lowers into new f64 vec
- Emplace uppers into the existing f64 vec

In fact, the new signature would prove useful for the general case as well to facilitate consolidating data containers